src/thunderbots/software/backend_output/grsim/motion_controller.cpp

```diff
@@ -0,0 +1,182 @@
+//
+// Created by evan on 18/08/18.
+//
+
+#include "motion_controller.h"
+
+#include <chrono>
+#include <ctime>
+#include <iostream>
+#include <utility>
+
+#include "ai/world/robot.h"
+#include "geom/angle.h"
+#include "geom/point.h"
+#include "shared/constants.h"
+
+
+// .cpp file for the grSim motion controller.
+//
+// In the current state it is a bang-bang controller.
+//
+// It assumed the robot max acceleration is constant.
+//
+// Uses constant acceleration kinematics equations to
+// calculate changes in speed.
+
+std::pair<Vector, Angle> MotionController::grSimBangBang(
+    Robot robot, Point dest, const double desired_final_speed,
+    const Angle desired_final_orientation, double delta_time)
+{
+    Vector robot_linear_velocities;  // vector to hold the XY velocities of the robot
+    bool b_can_stop_in_time;  // boolean value if the robot can reach it's destination at
+                              // target speed based on linear MAX acceleration
+    bool b_can_stop_rotate_in_time;  // boolean value if the robot can reach it's
+                                     // destination at target speed based on angular MAX
+                                     // acceleration
+
+    double robot_angular_velocity;
+    const double distance_to_dest =
+        (robot.position() - dest).len();  // destination distance used for constant linear
+                                          // acceleration speed calculations
+    const double angle_to_dest =
```

Comment:
There's an `Angle::diff` function for this, unless you also care about direction (pos vs neg)

Reply:
Changed controller entirely, no longer relevant